logos_setup.py

```diff
@@ -0,0 +1,801 @@
+        logos_info("Creating binary list.")
+        binaries = createWineBinaryList()
+        WINEBIN_OPTIONS = getWineBinOptions(binaries)
+
+        FLPRODUCT=os.environ.get("FLPRODUCT", "")
+        LOGOS_VERSION=os.environ.get("LOGOS_VERSION", "")
+        INSTALLDIR=os.environ.get("INSTALLDIR", "")
+        BACKTITLE="Choose Wine Binary Menu"
+        TITLE="Choose Wine Binary"
+        QUESTION_TEXT=f"Which Wine binary and install method should the script use to install {FLPRODUCT} v{LOGOS_VERSION} in {INSTALLDIR}?"
+
+        if os.environ.get('DIALOG') in ["whiptail", "dialog", "curses"]:
+            installationChoice = curses_menu(WINEBIN_OPTIONS, TITLE, QUESTION_TEXT)
+            WINECHOICE_CODE = installationChoice[0]
+            WINE_EXE = installationChoice[1]
+        elif os.environ["DIALOG"] == "zenity":
+            column_names = ["--column", "Choice", "--column", "Code", "--column", "Description", "--column", "Path"]
+            installationChoice = subprocess.check_output(["zenity", "--width=1024", "--height=480", "--title=" + TITLE, "--text=" + QUESTION_TEXT, "--list", "--radiolist"] + column_names + WINEBIN_OPTIONS + ["--print-column=2,3,4"]).decode().strip()
+            installArray = installationChoice.split('|')
+            WINECHOICE_CODE = installArray[0]
+            WINE_EXE = installArray[2]
+        elif os.environ["DIALOG"] == "kdialog":
+            no_diag_msg("kdialog not implemented.")
+        else:
+            no_diag_msg("No dialog tool found.")
+
+        os.environ['WINEBIN_CODE'] = WINECHOICE_CODE
+        os.environ['WINE_EXE'] = WINE_EXE
+        if VERBOSE:
+            print(f"chooseInstallMethod(): WINEBIN_CODE: {WINECHOICE_CODE}; WINE_EXE: {WINE_EXE}")
+
+def checkExistingInstall():
+    # Now that we know what the user wants to install and where, determine whether an install exists and whether to continue.
+    INSTALLDIR = os.environ.get('INSTALLDIR')
+    VERBOSE = os.getenv('VERBOSE')
+    if os.path.isdir(INSTALLDIR):
+        if VERBOSE:
+            print(f"INSTALLDIR: {INSTALLDIR}")
+        drive_c = f"{INSTALLDIR}/data/wine64_bottle/drive_c" # FIXME: Is WINEPREFIX already known here?
+        names = ['Logos.exe', 'Verbum.exe']
+        if os.path.isdir(drive_c) and any(glob.glob(f"{drive_c}/**/{n}", recursive=True) for n in names):
+            global EXISTING_LOGOS_INSTALL
+            EXISTING_LOGOS_INSTALL = 1
+            print(f"An install was found at {INSTALLDIR}. Please remove/rename it or use another location by setting the INSTALLDIR variable.")
+            return True
+        else:
+            global EXISTING_LOGOS_DIRECTORY
+            EXISTING_LOGOS_DIRECTORY = 1
+            print(f"A directory exists at {INSTALLDIR}. Please remove/rename it or use another location by setting the INSTALLDIR variable.")
+            return True
+    else:
+        if VERBOSE:
+            print(f"Installing to an empty directory at {INSTALLDIR}.")
+
+def beginInstall():
```

Comment:
Would it be better to have a cli.py over logos_setup.py, and then have logos_setup.py call either cli.py or gui/app.py as needed?